Tasks/PublishSymbols/Helpers.ps1

```diff
@@ -97,25 +59,26 @@ function Add-DbghelpLibrary {
 function Add-SourceServerStream {
     [cmdletbinding()]
     param(
-        [string]$PdbStrPath = $(throw 'Missing PdbStrPath.'),
-        [string]$SymbolsFilePath = $(throw 'Missing SymbolsFilePath.'),
-        [string]$StreamContent = $(throw 'Missing StreamContent.')
+        [ValidateNotNullOrEmpty()]
+        [string]$PdbStrPath,
+
+        [ValidateNotNullOrEmpty()]
+        [string]$SymbolsFilePath,
+
+        [ValidateNotNullOrEmpty()]
+        [string]$StreamContent
     )
 
     # Create a temp file to store the stream content.
     Write-Verbose "Writing source server stream content to a temp file."
-    [string]$streamContentFilePath = [System.IO.Path]::GetTempFileName()
+    $streamContentFile = New-TemporaryFile
     try {
-        # Check for spaces in the temp file path.
-        if ($streamContentFilePath.Contains(' ')) {
-            Write-Warning (Get-LocalizedString -Key 'Source files may not be indexed properly. Temp folder contains spaces in the path. To fix the issue, set the temp folder to a path without spaces.')
```

Comment:
This warning moved to a different place. We don't need to print it for every DLL.